platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/initialization/ConfigurationCacheStartParameter.kt

```diff
@@ -74,19 +74,29 @@ class ConfigurationCacheStartParameter internal constructor(
      */
     val isDeduplicatingStrings: Boolean = options.getInternalFlag("org.gradle.configuration-cache.internal.deduplicate-strings", true)
 
+    /**
+     * Whether configuration cache storing/loading should be done in parallel.
+     *
+     * Same as [StartParameterInternal.configurationCacheParallel].
+     *
+     * @see StartParameterInternal.configurationCacheParallel
+     */
+    val isParallelCache: Boolean
+        get() = startParameter.isConfigurationCacheParallel
+
     /**
      * Whether configuration should be stored in parallel.
      *
-     * The default is `true`.
+     * The default is the value of [isParallelCache].
      */
-    val isParallelStore = options.getInternalFlag("org.gradle.configuration-cache.internal.parallel-store", true)
+    val isParallelStore = isParallelCache && options.getInternalFlag("org.gradle.configuration-cache.internal.parallel-store", true)
 
     /**
      * Whether configuration should be loaded in parallel.
      *
-     * The default is `true`.
+     * The default is the value of [isParallelCache].
      */
-    val isParallelLoad = options.getInternalFlag("org.gradle.configuration-cache.internal.parallel-load", true)
+    val isParallelLoad = isParallelCache && options.getInternalFlag("org.gradle.configuration-cache.internal.parallel-load", true)
```

Comment:
❌ Parallel load shouldn't be affected by `isParallelCache`

Reply:
I guess I was excessively cautious.